src/hbbft_cc.erl

```diff
@@ -162,22 +146,17 @@ deserialize(#cc_serialized_data{state = State, sid = SID, n = N, f = F, shares =
         shares = deserialize_shares(hbbft_utils:curve(SK), SK, Shares)
     }.
 
--spec serialize_shares('BLS12-381' | 'SS512', #{non_neg_integer() => tc_signature_share:sig_share()}) -> #{non_neg_integer() => binary()}.
+-spec serialize_shares('BLS12-381', #{non_neg_integer() => tc_signature_share:sig_share()}) -> #{non_neg_integer() => binary()}.
 serialize_shares(Curve, Shares) ->
     maps:map(fun(_K, V) -> hbbft_utils:sig_share_to_binary(Curve, V) end, Shares).
 
--spec deserialize_shares('BLS12-381' | 'SS512', tc_key_share:tc_key_share() | tpke_privkey:privkey(), #{non_neg_integer() => binary()}) -> #{non_neg_integer() => tc_signature_share:sig_share()}.
+-spec deserialize_shares('BLS12-381', tc_key_share:tc_key_share(), #{non_neg_integer() => binary()}) -> #{non_neg_integer() => tc_signature_share:sig_share()}.
 deserialize_shares(Curve, SK, Shares) ->
     maps:map(fun(_K, V) -> hbbft_utils:binary_to_sig_share(Curve, SK, V) end, Shares).
 
-serialize_sid(SID) when is_binary(SID) ->
-    SID;
-serialize_sid(SID) ->
-    erlang_pbc:element_to_binary(SID).
+serialize_sid(<<SID/binary>>) ->
+    SID.
 
 deserialize_sid(SK, SID) ->
-    case hbbft_utils:curve(SK) of
-        'BLS12-381' -> SID;
-        'SS512' -> tpke_pubkey:deserialize_element(tpke_privkey:public_key(SK), SID)
-    end.
-
+    'BLS12-381' = hbbft_utils:curve(SK),
```

Comment:
I suppose this is fine now that mainnet would never be in a transitional phase? Maybe add a comment saying so? Although I do wonder whether we even need to [de]ser this `SID` since its just transparent.

Reply:
True - this is completely redundant - removed.